components/calendar/src/calendar_arithmetic.rs

```diff
@@ -279,6 +279,43 @@ impl<C: CalendarArithmetic> ArithmeticDate<C> {
 
         Ok(Self::new_unchecked(year, month, day))
     }
+
+    /// Construct a new arithmetic date from a year, month code, and day, for a lunar calendar
+    /// with intercalary months; see [`new_from_solar_codes`]
+    pub fn new_from_lunar_codes<C2: Calendar>(
```

Comment:
issue: i don't think this method adds value since it's going to have to be designed individually per calendar

Reply:
I do think it is nice to have a centralized place for the month/day range validation logic. It was scattered everywhere before and I centralized it into `new_from_solar_codes` and `new_from_solar_ordinals`.